pkg/models/user_info.go

```diff
@@ -8,3 +8,10 @@ type UserInfo struct {
 	Email       EmailAddress
 	Username    string
 }
+
+func (ui UserInfo) GetMappingKey() string {
+	return ui.Username
+}
+func (ui UserInfo) GetMappingVal() *UserInfo {
+	return &ui
+}
```

Comment:
This might apply to most of these functions, but I'd be curious to see the pros/cons of modifying these `GetMappingVal` functions to instead look like this:

```suggestion
func (ui *UserInfo) GetMappingVal() *UserInfo {
	return ui
}
```

and instead just operate on the pointer type. This prevents copies of the struct from being made, and might be a bit more true to the expected behavior

https://goplay.tools/snippet/8zm_YC0blq8

If changing this causes a bunch of hellish changes throughout the codebase, don't worry about it, but if we can just easily flip the receiver type then I think it might be worth it!

Thoughts, @mynar7 / @samoddball / @StevenWadeOddball?

Reply:
I had done a [little research](https://google.github.io/styleguide/go/decisions#receiver-type) because I had the same question originally.

I'll say that because we're only using `GetMappingVal` in dataloaders to return the inner value (itself a simple struct) that in practice it shouldn't matter whether the method is attaching to the struct or a pointer.

Stylistically, I opted to not attach directly to the struct just because I thought it might be simpler (I thought this might eliminate nil checking) and I wanted all of these methods to be consistent. That said, I'm open to flipping them to pointers if I'm misunderstanding something or someone feels strongly about changing these.

Reply:
Nah, not strongly enough right now. Can be easily changed later, and I really don't think it's got any practical impact on much right now.